tkg/client/audit_upgrade_helper_test.go

```diff
@@ -0,0 +1,114 @@
CAgIHJlc291cmNlczoKICAgIC0gbmFtZXNwYWNlcwogICAgLSBuYW1lc3BhY2VzL3N0YXR1cwogICAgLSBuYW1lc3BhY2VzL2ZpbmFsaXplCi0gbGV2ZWw6IE5vbmUKICB1c2VyczoKICAtIHN5c3RlbTprdWJlLWNvbnRyb2xsZXItbWFuYWdlcgogIHZlcmJzOgogIC0gZ2V0CiAgLSBsaXN0CiAgcmVzb3VyY2VzOgogIC0gZ3JvdXA6IG1ldHJpY3MuazhzLmlvCi0gbGV2ZWw6IE5vbmUKICBub25SZXNvdXJjZVVSTHM6CiAgLSAvaGVhbHRoeioKICAtIC92ZXJzaW9uCiAgLSAvc3dhZ2dlcioKLSBsZXZlbDogTm9uZQogIHJlc291cmNlczoKICAtIGdyb3VwOiAiIgogICAgcmVzb3VyY2VzOgogICAgLSBldmVudHMKLSBsZXZlbDogTm9uZQogIHVzZXJHcm91cHM6CiAgLSBzeXN0ZW06c2VydmljZWFjY291bnRzOnZtd2FyZS1zeXN0ZW0tdG1jCiAgdmVyYnM6CiAgLSBnZXQKICAtIGxpc3QKICAtIHdhdGNoCi0gbGV2ZWw6IE5vbmUKICB1c2VyczoKICAtIHN5c3RlbTpzZXJ2aWNlYWNjb3VudDprdWJlLXN5c3RlbTpnZW5lcmljLWdhcmJhZ2UtY29sbGVjdG9yCiAgdmVyYnM6CiAgLSBnZXQKICAtIGxpc3QKICAtIHdhdGNoCi0gbGV2ZWw6IFJlcXVlc3QKICB1c2VyR3JvdXBzOgogIC0gc3lzdGVtOm5vZGVzCiAgdmVyYnM6CiAgLSB1cGRhdGUKICAtIHBhdGNoCiAgcmVzb3VyY2VzOgogIC0gZ3JvdXA6ICIiCiAgICByZXNvdXJjZXM6CiAgICAtIG5vZGVzL3N0YXR1cwogICAgLSBwb2RzL3N0YXR1cwogIG9taXRTdGFnZXM6CiAgLSBSZXF1ZXN0UmVjZWl2ZWQKLSBsZXZlbDogUmVxdWVzdAogIHVzZXJzOgogIC0gc3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOm5hbWVzcGFjZS1jb250cm9sbGVyCiAgdmVyYnM6CiAgLSBkZWxldGVjb2xsZWN0aW9uCiAgb21pdFN0YWdlczoKICAtIFJlcXVlc3RSZWNlaXZlZAotIGxldmVsOiBNZXRhZGF0YQogIHJlc291cmNlczoKICAtIGdyb3VwOiAiIgogICAgcmVzb3VyY2VzOgogICAgLSBzZWNyZXRzCiAgICAtIGNvbmZpZ21hcHMKICAtIGdyb3VwOiBhdXRoZW50aWNhdGlvbi5rOHMuaW8KICAgIHJlc291cmNlczoKICAgIC0gdG9rZW5yZXZpZXdzCiAgb21pdFN0YWdlczoKICAtIFJlcXVlc3RSZWNlaXZlZAotIGxldmVsOiBSZXF1ZXN0CiAgdmVyYnM6CiAgLSBnZXQKICAtIGxpc3QKICAtIHdhdGNoCiAgcmVzb3VyY2VzOgogIC0gZ3JvdXA6ICIiCiAgLSBncm91cDogYWRtaXNzaW9ucmVnaXN0cmF0aW9uLms4cy5pbwogIC0gZ3JvdXA6IGFwaWV4dGVuc2lvbnMuazhzLmlvCiAgLSBncm91cDogYXBpcmVnaXN0cmF0aW9uLms4cy5pbwogIC0gZ3JvdXA6IGFwcHMKICAtIGdyb3VwOiBhdXRoZW50aWNhdGlvbi5rOHMuaW8KICAtIGdyb3VwOiBhdXRob3JpemF0aW9uLms4cy5pbwogIC0gZ3JvdXA6IGF1dG9zY2FsaW5nCiAgLSBncm91cDogYmF0Y2gKICAtIGdyb3VwOiBjZXJ0aWZpY2F0ZXMuazhzLmlvCiAgLSBncm91cDogZXh0ZW5zaW9ucwogIC0gZ3JvdXA6IG1ldHJpY3MuazhzLmlvCiAgLSBncm91cDogbmV0d29ya2luZy5rOHMuaW8KICAtIGdyb3VwOiBwb2xpY3kKICAtIGdyb3VwOiByYmFjLmF1dGhvcml6YXRpb24uazhzLmlvCiAgLSBncm91cDogc2V0dGluZ3MuazhzLmlvCiAgLSBncm91cDogc3RvcmFnZS5rOHMuaW8KICBvbWl0U3RhZ2VzOgogIC0gUmVxdWVzdFJlY2VpdmVkCi0gbGV2ZWw6IFJlcXVlc3RSZXNwb25zZQogIHJlc291cmNlczoKICAtIGdyb3VwOiAiIgogIC0gZ3JvdXA6IGFkbWlzc2lvbnJlZ2lzdHJhdGlvbi5rOHMuaW8KICAtIGdyb3VwOiBhcGlleHRlbnNpb25zLms4cy5pbwogIC0gZ3JvdXA6IGFwaXJlZ2lzdHJhdGlvbi5rOHMuaW8KICAtIGdyb3VwOiBhcHBzCiAgLSBncm91cDogYXV0aGVudGljYXRpb24uazhzLmlvCiAgLSBncm91cDogYXV0aG9yaXphdGlvbi5rOHMuaW8KICAtIGdyb3VwOiBhdXRvc2NhbGluZwogIC0gZ3JvdXA6IGJhdGNoCiAgLSBncm91cDogY2VydGlmaWNhdGVzLms4cy5pbwogIC0gZ3JvdXA6IGV4dGVuc2lvbnMKICAtIGdyb3VwOiBtZXRyaWNzLms4cy5pbwogIC0gZ3JvdXA6IG5ldHdvcmtpbmcuazhzLmlvCiAgLSBncm91cDogcG9saWN5CiAgLSBncm91cDogcmJhYy5hdXRob3JpemF0aW9uLms4cy5pbwogIC0gZ3JvdXA6IHNldHRpbmdzLms4cy5pbwogIC0gZ3JvdXA6IHN0b3JhZ2UuazhzLmlvCiAgb21pdFN0YWdlczoKICAtIFJlcXVlc3RSZWNlaXZlZAotIGxldmVsOiBNZXRhZGF0YQogIG9taXRTdGFnZXM6CiAgLSBSZXF1ZXN0UmVjZWl2ZWQK"
```

Comment:
It would be easy to understand the test case or make any changes with plain text auditconfig when we revisit it someday. We can encode it in test method.

Reply:
yup, we did not catch the release cut, will have time to polish these, thanks